pages/contribution/contribution-modal.vue

```diff
@@ -30,284 +34,240 @@
-			}
-
-			this.$set(this, "formRecords", {
-				[createdFormObj.formId]: createdFormObj,
+			this.$nextTick(() => {
+				this.selectedContrib = newLen - 1;
 			});
-			this.openedFormId = createdFormObj.formId;
-			this.closeOnSubmit = !!closeOnSubmit;
 		},
-		contributorsFromIds(authorIds) {
-			if (!authorIds || authorIds.length === 0) return "";
-
-			const contributorNames = this.contributors
-				.filter((c) => authorIds.includes(c.id))
-				.map((c) => c.username);
+		deleteContrib(index) {
+			// do not continue if not found
+			if (!this.contribs[index]) return;
 
-			const total = contributorNames.length;
-			const anonymousTotal = contributorNames.filter((username) => !username).length;
-			const knownNames = contributorNames.filter((username) => username);
+			// do not delete if only one
+			if (this.contribs.length <= 1) return;
 
-			if (anonymousTotal > 0) {
-				const anonymousStr = `${anonymousTotal} ${this.$root.lang().database.anonymous}`;
-				knownNames.splice(0, 0, anonymousStr);
-			}
+			this.contribs.splice(index, 1);
 
-			return `[${total}]: ${knownNames.join(", ")}`;
+			// if we deleted the opened form, change it to the previous one
+			if (this.contribs.length === this.selectedContrib) --this.selectedContrib;
 		},
-		changeOpenedForm(formId) {
-			if (this.openedFormId === formId) this.openedFormId = undefined;
-			else this.openedFormId = formId;
+		changeOpenedForm(index) {
+			this.selectedContrib = index;
 		},
 		close() {
-			this.modalOpened = false;
+			this.$emit("close", false);
 		},
-		closeAndCancel() {
-			this.close();
-			this.onCancel();
+		handleSubmit() {
+			// the code is different enough for adding/creating it's worth having two functions
+			return this.add ? this.addContributions() : this.editContribution();
 		},
-		async closeOrAndSubmit() {
-			const resultDataList = Object.values(this.formRecords).map((f) => {
-				delete f.formId;
-				return f;
-			});
+		async addContributions() {
+			let success = true;
+			const finalContributions = [];
+			// can't use map since errors can be thrown
+			for (const contrib of this.contribs) {
+				// convert ranges into actual texture IDs
+				const generatedRange = generateRange(contrib.texture);
 
-			const dataPurified = JSON.parse(
-				JSON.stringify(this.multiple ? resultDataList : resultDataList[0]),
-			);
+				if (!generatedRange.length) {
+					this.$root
+						.jsonSnackBar(contrib)
+						.showSnackBar(
+							this.$root.lang().database.contributions.modal.id_field_errors.one_required,
+							"error",
+						);
+					console.error(contrib);
+					success = false;
+					break;
+				}
 
-			const wentWell = await this.onSubmit(dataPurified);
+				if (contrib.authors.length === 0) {
+					this.$root
+						.jsonSnackBar(contrib)
+						.showSnackBar(this.$root.lang().database.contributions.no_contributor_yet, "error");
+					console.error(contrib);
+					success = false;
+					break;
+				}
```

Comment:
Too much indentation levels
You should not use break, never disrupt code flow like a waterfall, it means your code isn't good enough
You must check all data before submission,

They should not be able to push any contribution if one is incorrect, all of them shall be good.
Use for loop to send as much snackbar as possible with all errors and only continue code for submission if all good